src/Swarm/TUI/Controller.hs

```diff
@@ -224,12 +224,17 @@ handleMainEvent s = \case
         continueWithoutRedraw s
       _ -> continueWithoutRedraw (s & uiState . uiWorldCursor .~ Nothing)
   MouseUp n _ _mouseLoc -> do
-    setFocus s $ case n of
+    let s' =
+          s & case n of
+            InventoryListItem pos -> uiState . uiInventory %~ maybe Nothing (\(h, l) -> Just (h, BL.listMoveTo pos l))
```

Comment:
There is surely a better lense to update the second element of a maybe tuple, but I couldn't find it :)

Reply:
How's that?  `traverse` is to a lens as `fmap` is to applying a function.

Reply:
:exploding_head: